quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/config/TimefoldBuildTimeConfig.java

```diff
@@ -1,32 +1,47 @@
 package ai.timefold.solver.quarkus.deployment.config;
 
+import java.util.Map;
 import java.util.Optional;
 
 import ai.timefold.solver.core.config.solver.SolverConfig;
 
-import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithUnnamedKey;
 
 /**
  * During build time, this is translated into Timefold's Config classes.
  */
-@ConfigRoot(name = "timefold")
-public class TimefoldBuildTimeConfig {
+@ConfigMapping(prefix = "quarkus.timefold")
+@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
+public interface TimefoldBuildTimeConfig {
 
-    public static final String DEFAULT_SOLVER_CONFIG_URL = "solverConfig.xml";
+    String DEFAULT_SOLVER_CONFIG_URL = "solverConfig.xml";
+    String DEFAULT_SOLVER_NAME = "default";
 
     /**
      * A classpath resource to read the solver configuration XML.
      * Defaults to {@value DEFAULT_SOLVER_CONFIG_URL}.
      * If this property isn't specified, that solverConfig.xml is optional.
      */
-    @ConfigItem
-    public Optional<String> solverConfigXml;
+    Optional<String> solverConfigXml();
 
     /**
-     * Configuration properties that overwrite Timefold's {@link SolverConfig}.
+     * Configuration properties that overwrite Timefold's {@link SolverConfig} per Solver. If a solver name is not
+     * explicitly specified, the solver name will default to {@link #DEFAULT_SOLVER_NAME}.
```

Comment:
```suggestion
     * Configuration properties that overwrite the {@link SolverConfig} per Solver. 
     * If a solver name is not explicitly specified, the solver name will default to {@link #DEFAULT_SOLVER_NAME}.
```

When writing about Timefold, please remember:

- Timefold is the company.
- Timefold Solver is the product.

In this case, no need to mention Timefold at all; everyone knows that it's inside the Timefold Solver code.